File: z-clients/python/diom/apis/cache_namespace.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    CacheConfigureNamespaceIn,
    CacheConfigureNamespaceOut,
    CacheGetNamespaceIn,
    CacheGetNamespaceOut,
)


class CacheNamespaceAsync(ApiBase):
    async def configure(
        self,
        cache_configure_namespace_in: CacheConfigureNamespaceIn,
    ) -> CacheConfigureNamespaceOut:
        """Configure cache namespace"""
        body = cache_configure_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.cache.namespace.configure",
            body=body,
            response_type=CacheConfigureNamespaceOut,
        )

    async def get(
        self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> CacheGetNamespaceOut:
        """Get cache namespace"""
        body = cache_get_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.cache.namespace.get",
            body=body,
            response_type=CacheGetNamespaceOut,
        )


class CacheNamespace(ApiBase):
    def configure(
        self,
        cache_configure_namespace_in: CacheConfigureNamespaceIn,
    ) -> CacheConfigureNamespaceOut:
        """Configure cache namespace"""
        body = cache_configure_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.cache.namespace.configure",
            body=body,
            response_type=CacheConfigureNamespaceOut,
        )

    def get(
        self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> CacheGetNamespaceOut:
        """Get cache namespace"""
        body = cache_get_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.cache.namespace.get",
            body=body,
            response_type=CacheGetNamespaceOut,
        )

File: z-clients/rust/src/api/cache_namespace.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct CacheNamespace<'a> {
    cfg: &'a Configuration,
}

impl<'a> CacheNamespace<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Configure cache namespace
    pub async fn configure(
        &self,
        cache_configure_namespace_in: CacheConfigureNamespaceIn,
    ) -> Result<CacheConfigureNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.cache.namespace.configure")
            .with_body(cache_configure_namespace_in)
            .execute(self.cfg)
            .await
    }

    /// Get cache namespace
    pub async fn get(
        &self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> Result<CacheGetNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.cache.namespace.get")
            .with_body(cache_get_namespace_in)
            .execute(self.cfg)
            .await
    }
}
